hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/ResourceCalculationDriver.java

```diff
@@ -20,64 +20,63 @@
+ * bookkeeper of disposable update information that is used by all children under the common parent.
  */
 public class ResourceCalculationDriver {
-  protected static final Set<ResourceUnitCapacityType> CALCULATOR_PRECEDENCE =
-      ImmutableSet.of(
+  protected static final ResourceUnitCapacityType[] CALCULATOR_PRECEDENCE =
+      new ResourceUnitCapacityType[] {
           ResourceUnitCapacityType.ABSOLUTE,
           ResourceUnitCapacityType.PERCENTAGE,
-          ResourceUnitCapacityType.WEIGHT);
+          ResourceUnitCapacityType.WEIGHT};
+  static final String MB_UNIT = "Mi";
 
   protected final QueueResourceRoundingStrategy roundingStrategy =
       new DefaultQueueResourceRoundingStrategy(CALCULATOR_PRECEDENCE);
-  protected final CSQueue parent;
+  protected final CSQueue queue;
   protected final QueueCapacityUpdateContext updateContext;
   protected final Map<ResourceUnitCapacityType, AbstractQueueCapacityCalculator> calculators;
   protected final Collection<String> definedResources;
 
-  protected final Map<String, ResourceVector> overallRemainingResource = new HashMap<>();
-  protected final Map<String, ResourceVector> batchRemainingResource = new HashMap<>();
+  protected final Map<String, ResourceVector> overallRemainingResourcePerLabel = new HashMap<>();
+  protected final Map<String, ResourceVector> batchRemainingResourcePerLabel = new HashMap<>();
   // Used by ABSOLUTE capacity types
-  protected final Map<String, ResourceVector> normalizedResourceRatio = new HashMap<>();
-  // Used by WEIGHT capacity typet js
-  protected final Map<String, Map<String, Float>> sumWeightsPerLabel = new HashMap<>();
-
-  protected String currentResourceName;
-  protected AbstractQueueCapacityCalculator currentCalculator;
-  protected CSQueue currentChild;
-  protected Map<String, Float> usedResourceByCurrentCalculator = new HashMap<>();
+  protected final Map<String, ResourceVector> normalizedResourceRatioPerLabel = new HashMap<>();
+  // Used by WEIGHT capacity types
+  protected final Map<String, Map<String, Double>> sumWeightsPerLabel = new HashMap<>();
+  protected Map<String, Double> usedResourceByCurrentCalculatorPerLabel = new HashMap<>();
 
   public ResourceCalculationDriver(
-      CSQueue parent, QueueCapacityUpdateContext updateContext,
+      CSQueue queue, QueueCapacityUpdateContext updateContext,
       Map<ResourceUnitCapacityType, AbstractQueueCapacityCalculator> calculators,
       Collection<String> definedResources) {
-    this.parent = parent;
+    this.queue = queue;
     this.updateContext = updateContext;
     this.calculators = calculators;
     this.definedResources = definedResources;
+    setNormalizedResourceRatio();
   }
 
+
   /**
    * Returns the parent that is driving the calculation.
    *
    * @return a common parent queue
    */
-  public CSQueue getParent() {
-    return parent;
+  public CSQueue getQueue() {
+    return queue;
   }
 
```

Comment:
You might consider adding a helper getChildren method as well, you have a lot of resourceCalculator.getQueue().getChildren() calls, you could proxy the queue.getChilder() as resourceCalculator.getChildren(). It's just a suggestion, it's fine this way as well.